services/CollectionService/src/index.php

```diff
@@ -67,40 +49,127 @@
-   $subRequest = Request::create($url, 'GET', array(), $request->cookies->all(), $request->files->all(), $request->server->all());
-   error_log($subRequest->__toString());
-   $response = $app->handle($subRequest, HttpKernelInterface::SUB_REQUEST, false);
-   return $response;
+$app->post("/islandora/collection/{id}", function (Request $request, $id) use ($app) {
+  $tx = $request->query->get('tx', "");
+
+  //Check for format
+  $format = NULL;
+  try {
+    $format = \EasyRdf_Format::getFormat($contentType = $request->headers->get('Content-Type', 'text/turtle'));
+  } catch (\EasyRdf_Exception $e) {
+    $app->abort(415, $e->getMessage());
+  }
+
+  //Now check if body can be parsed in that format
+  if ($format) { //EasyRdf_Format
+    //@see http://www.w3.org/2011/rdfa-context/rdfa-1.1 for defaults
+    \EasyRdf_Namespace::set('pcdm', 'http://pcdm.org/models#');
+    \EasyRdf_Namespace::set('nfo', 'http://www.semanticdesktop.org/ontologies/2007/03/22/nfo/v1.1/');
+    \EasyRdf_Namespace::set('isl', 'http://www.islandora.ca/ontologies/2016/02/28/isl/v1.0/');
+    \EasyRdf_Namespace::set('ldp', 'http://www.w3.org/ns/ldp');
+
+    //Fake IRI, default LDP one for current resource "<>" is not a valid IRI!
+    $fakeIri = new \EasyRdf_ParsedUri('urn:uuid:'.$app['uuid5']);
+
+    $graph = new \EasyRdf_Graph();
+    try {
+      $graph->parse($request->getContent(), $format->getName(), $fakeIri);
+    } catch (\EasyRdf_Exception $e) {
+      $app->abort(415, $e->getMessage());
+    }
+    //Add a pcmd:Collection type
+    $graph->resource($fakeIri, 'pcdm:Collection');
+
+    //Check if we got an UUID inside posted RDF. We won't validate it here because it's the caller responsability
+    if (NULL != $graph->countValues($fakeIri, '<http://www.semanticdesktop.org/ontologies/2007/03/22/nfo/v1.1/uuid>')) {
+      $existingUuid = $graph->getLiteral($fakeIri, '<http://www.semanticdesktop.org/ontologies/2007/03/22/nfo/v1.1/uuid>');
+      $graph->addResource($fakeIri, 'http://www.islandora.ca/ontologies/2016/02/28/isl/v1.0/hasURN', 'urn:uuid:'.$existingUuid); //Testing an Islandora Ontology!
+    } else {
+      //No UUID from the caller in RDF, lets put something there
+      $tmpUuid = $app['uuid']; //caching here, since it's regenerated each time it is used and I need the same twice
+      $graph->addLiteral($fakeIri,"http://www.semanticdesktop.org/ontologies/2007/03/22/nfo/v1.1/uuid",$tmpUuid); //Keeps compat for now with other services
+      $graph->addResource($fakeIri,"http://www.islandora.ca/ontologies/2016/02/28/isl/v1.0/hasURN",'urn:uuid:'.$tmpUuid); //Testing an Islandora Ontology
+    }
+    //Restore LDP <> IRI on serialised graph
+    $pcmd_collection_rdf= str_replace($fakeIri, '', $graph->serialise('turtle'));
+  }
+
+  $urlRoute = $request->getUriForPath('/islandora/resource/');
+ 
+  $subRequestPost = Request::create($urlRoute.$id, 'POST', array(), $request->cookies->all(), array(), $request->server->all(), $pcmd_collection_rdf);
```

Comment:
Lots of reasons, we get all the middleware (like checking if the base resource is there, header rewriting and we don't need to rewrite psr7 responses. More over, the idea is to not use chullo directly. ResourceService should deal with any new chullo functionality if needed (if we for example need to recode) but not in this higher level services. And since we already using all the other services ResourceService provides, going to chullo when we already have the routes  is sub-using our code.